### PR TITLE
Some Changes

### PR DESCRIPTION
**Pull request by UtsavSomaiya**
redirect + session for POST data in case of validation errors = half done(except unset)

categories - plural = done

store category_id in the products table = done 

delete product not working = done

### CODE REVIEW

File: admin/product/add_product.php [hunk 1]
```diff
@@ -2,46 +2,59 @@
...
+                    $_SESSION['msg'] = "Add Successfully";
+                    header('location:../product/show_product.php');
+                } else {
+                    $_SESSION['msg'] = "Not Successfully";
+                    header('location:../product/add_product.php');
+                }
+            }
+        } else {
+            if (empty($_POST['pname'])) {
+                $_SESSION['msg_alert1'] = "Please enter data..";
```
Comment (gauravmak) [inline admin/product/add_product.php:41]: $_SESSION['name_validation_error'] = "Please enter name";

File: admin/product/add_product.php [hunk 2]
```diff
@@ -60,13 +73,16 @@
 									placeholder="Product Name" name="pname"
                                     <?php
                                     if (isset($_POST['pname'])) {
-                                        echo "value=\"".$_POST['pname']."\"";
+                                        $_SESSION['pname']= $_POST['pname'];
+                                    }
+                                    if (isset($_SESSION['pname'])) {
+                                        echo "value=\"".$_SESSION['pname']."\"";
                                     }?>
-                                required>
```
Comment (gauravmak) [inline admin/product/add_product.php:65]: Not required?

File: admin/product/add_product.php [hunk 3]
```diff
@@ -77,40 +93,59 @@
...
+                                <select class="form-control" id="exampleSelectGender" name="category" >
                                     <option value="">--Select Category--</option>
                                     <?php
                                         $fetch = $pdo->prepare("select * from category");
                                         $fetch->execute();
                                         $res = $fetch->fetchAll();
                                         foreach ($res as $r1) {
                                             ?>
-                                    <option  value="<?= $r1['name'] ?>"><?= $r1['name']?></option>
+                                    <option  value="<?= $r1['id'] ?>"
```
Comment (gauravmak) [inline admin/product/add_product.php:121]: $r1 means?

File: admin/vendors/js/vendor.bundle.base.js [hunk 4]
```diff
@@ -1,7 +1,7 @@
 /*! jQuery v3.5.1 | (c) JS Foundation and other contributors | jquery.org/license */
 function alert(id) {
   if (confirm('Are you sure?')) {
-      window.location.href='/product/delete_product.php?id='+id;
+      window.location.href='/admin/product/delete_product.php?id='+id;
     }
 }
```
Comment (gauravmak) [inline admin/vendors/js/vendor.bundle.base.js:6]: Why add custom function in vendor file? Add your own

File: admin/product/show_product.php [hunk 5]
```diff
@@ -29,17 +29,26 @@
...
+									<td><?= "$".$r1['price'] ?></td>
+									<td><?php
+                                    $fetch = $pdo->prepare("select name from category where id = {$r1['category']}");
+                                            $fetch->execute();
+                                            $result = $fetch->fetchAll();
+                                            foreach ($result as $r) {
+                                                if (!empty($r)) {
+                                                    echo $r['name'];
+                                                }
+                                            } ?>
```
Comment (gauravmak) [inline admin/product/show_product.php:47]: Formatting?

Comment (gauravmak) [inline admin/product/show_product.php:47]: Formatting?

File: admin/product/add_product.php [hunk 6]
```diff
@@ -60,13 +73,16 @@
 									placeholder="Product Name" name="pname"
                                     <?php
                                     if (isset($_POST['pname'])) {
-                                        echo "value=\"".$_POST['pname']."\"";
+                                        $_SESSION['pname']= $_POST['pname'];
+                                    }
```
[the remark this answers is not in the document]
Comment (UtsavSomaiya) [reply admin/product/add_product.php:77]: it is store into the session so its value it is show on next page..

**Pull request merged**